src/components/app.js

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import React, { Component } from 'react';
+
+import { svgs } from 'utilities';
+import Shell from "components/shell/shell";
+import { ManageDeviceGroupsContainer, SettingsContainer } from 'components/shell/flyouts';
+import { DashboardContainer, DevicesContainer, RulesContainer, MaintenanceContainer } from './pages';
+
+class App extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = { openFlyout: '' };
+  }
+
+  closeFlyout = () => this.setState({ openFlyout: '' });
+
+  openSettings = () => this.setState({ openFlyout: 'settings' });
+
+  render() {
+    const { deviceGroupFlyoutIsOpen } = this.props;
+    const { openFlyout } = this.state;
+
+    const pagesConfig = [
+      {
+        to: '/dashboard',
+        exact: true,
+        svg: svgs.tabs.dashboard,
+        labelId: 'tabs.dashboard',
+        component: DashboardContainer
+      },
+      {
+        to: '/devices',
+        exact: true,
+        svg: svgs.tabs.devices,
+        labelId: 'tabs.devices',
+        component: DevicesContainer
+      },
+      {
+        to: '/rules',
+        exact: true,
+        svg: svgs.tabs.rules,
+        labelId: 'tabs.rules',
+        component: RulesContainer
+      },
+      {
+        to: '/maintenance',
+        exact: false,
+        svg: svgs.tabs.maintenance,
+        labelId: 'tabs.maintenance',
+        component: MaintenanceContainer
+      }
+    ];
+
+    const crumbsConfig = [
+      {
```

Comment:
curmbsConfig should probably be combined with the pagesConfig

Reply:
I really struggled with how to structure the breadcrumbs... and whether to combine with the pagesConfig.  For the top level addresses, they are very related and I didn't like the redundancy. However, for the other pages like "/maintenance/rule/:id" it felt odd to put those into pagesConfig with a boolean for includeInNav (or something similar).